### PR TITLE
Add pkg/version

### LINKED ISSUE #26: registry image reports "unknown" openshift version
time="2017-12-11T18:50:48.773847179Z" level=info msg="start registry" distribution_version="v2.6.2+unknown" go.version=go1.8.3 instance.id=0bfa960a-ac57-4138-abf9-f9445aa5eed7 kubernetes_version=v1.7.6+$Format:%h$ openshift_version=unknown

### PR DESCRIPTION
**Pull request by dmage**
/assign @bparees 
Resolves #26

### CODE REVIEW

**Comment by bparees** [comment]
lgtm but is our build process actually populating those fields?

File: pkg/version/version.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package version
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	// commitFromGit is a constant representing the source version that
+	// generated this build. It should be set during build via -ldflags.
```
Comment (bparees) [inline pkg/version/version.go]: that is, is -ldflags being set in our build process?

Comment (dmage) [reply pkg/version/version.go]: Yes, https://github.com/openshift/image-registry/blob/c540fa0224d2028b9633a147c283fbf60d371ba9/hack/lib/constants.sh#L59-L63

Comment (bparees) [reply pkg/version/version.go]: cool, thanks.

**Comment by dmage** [comment]
@bparees looks like the extended test is broken.

**Comment by bparees** [comment]
> @bparees looks like the extended test is broken.

or flaked anyway. are you looking into it?

**Comment by dmage** [comment]
issues for extended tests: origin/issues#17786

**Comment by dmage** [comment]
issue for extended tests: origin/issues#17786

**Comment by dmage** [comment]
issue for extended tests: openshift/origin#17786

**Pull request merged**